src/test/java/com/volasoftware/tinder/services/implementations/FriendServiceImplTest.java

```diff
@@ -308,6 +306,39 @@ public void testSortFriendsByLocation() {
         assertEquals(expectedFriendDtos, sortedFriendDtos);
     }
 
+    @Test
+    void testGetFriendInfoWhenFriendExistThenReturnFriendDto() {
+        //given
+        String email = "jacob@gmail.com";
+        Long friendId = 1L;
+
+        Account loggedAccount = new Account();
+        loggedAccount.setEmail(email);
+        loggedAccount.setFriends(new HashSet<>());
+
+        Account friend = new Account();
+        friend.setId(friendId);
+        loggedAccount.getFriends().add(friend);
+
+        FriendDto friendDto = new FriendDto();
+        friendDto.setFirstName("Jacob");
+
+        //when
+        when(accountRepository.findOneByEmail(email)).thenReturn(Optional.of(loggedAccount));
+        when(accountRepository.findById(friendId)).thenReturn(Optional.of(friend));
+        when(friendMapper.accountToFriendDto(friend)).thenReturn(friendDto);
+
+        // Set the security context
+        Authentication auth = new UsernamePasswordAuthenticationToken(email, null);
+        SecurityContextHolder.getContext().setAuthentication(auth);
```

Comment:
Do you need that in the test?

Reply:
Why is added now and missing in previous created tests in this class about removing and adding a friend, because the account's retrieve logic is the same?